drivers/power/adp1050/adp1050.c

```diff
@@ -0,0 +1,1252 @@
+ * @brief Set ADP1050 VOUT transition rate
+ * @param desc - ADP1050 device descriptor
+ * @param vout_tr - VOUT transition rate value to be set.
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int adp1050_vout_tr(struct adp1050_desc *desc, enum adp1050_vout_tr vout_tr)
+{
+	return adp1050_write(desc, ADP1050_VOUT_TRANSITION_RATE, vout_tr, 2);
+}
+
+/**
+ * @brief Set ADP1050 VOUT scale, regarding the resistor divider assigned to
+ * 	  the VS pins
+ * @param desc - ADP1050 device descriptor
+ * @param exp - Exponent - 5-bit twos complement
+ * @param mant - Mantissa - 11-bit twos complement
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int adp1050_vout_scale(struct adp1050_desc *desc, int8_t exp, uint16_t mant)
+{
+	int ret;
+	uint16_t vout_scale;
+
+	if (exp > ADP1050_EXP_MAX || mant > ADP1050_MANT_MAX)
+		return -EINVAL;
+
+	vout_scale = no_os_field_prep(ADP1050_EXP_MASK, exp) | mant;
+
+	return adp1050_write(desc, ADP1050_VOUT_SCALE_MONITOR, vout_scale, 2);
+}
+
+/**
+ * @brief Set VOUT margins value
+ * @param desc - ADP1050 device descriptor
+ * @param margin_high - VOUT margin high value (Y value for V = Y x 2^N format),
+ * 			N defined in VOUT_MODE (-10).
+ * @param margin_low - VOUT margin low value (Y value for V = Y x 2^N format),
+ * 		       N defined in VOUT_MODE (-10).
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int adp1050_vout_margin(struct adp1050_desc *desc, uint16_t margin_high,
+			uint16_t margin_low)
+{
+	int ret;
+
+	ret = adp1050_write(desc, ADP1050_VOUT_MARGIN_HIGH, margin_high, 2);
+	if (ret)
+		return ret;
+
+	return adp1050_write(desc, ADP1050_VOUT_MARGIN_LOW, margin_low, 2);
+}
+
+/**
+ * @brief Set VOUT duty cycle for requested channel
+ * @param desc - ADP1050 device descriptor
+ * @param pulse_width - Width of the pulse wanted for the signal (used for
+ * 			calculation of the 12-bit rising and falling edge
+ * 			timings for the requested channels, it depends on each
+ * 			channel how the falling and rising edges are calculated
+ * 			using this value). Each LSB corresponds to a 5ns
+ * 			resolution.
+ * @param pulse_start - Start of the pulse wanted for the signal (used for
+ * 			calculation of the 12-bit rising and falling edge
+ * 			timings for the requested channels, it depends on each
+ * 			channel how the falling and rising edges are calculated
+ * 			using this value). Each LSB corresponds to a 5ns
+ * 			resolution.
+ * @param chan - Requested channel.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int adp1050_pwm_duty_cycle(struct adp1050_desc *desc, uint16_t pulse_width,
+			   uint16_t pulse_start, enum adp1050_channel chan)
+{
+	int ret;
+	uint8_t redge_msb, fedge_msb, lsb;
+
+	if (no_os_field_get(ADP1050_FLOAT_FREQ_MASK, desc->freq))
+		if (pulse_width + pulse_start > ADP1050_MAX_PERIOD_FLOAT(desc->freq))
+			return -EINVAL;
+		else if (pulse_width + pulse_start > ADP1050_MAX_PERIOD_INTEGER(desc->freq))
```

Comment:
no need for else if you are returning in the previous if, also, you can use a single if statement with on or between which can actually merged with the first if condition with an and in between